.changeset/swift-fans-protect.md

```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/backend-common': minor
+---
+
+- Add `updateCorsOptions` method to merge new options with existent ones
```

Comment:
Sorry I might be out of the loop! I'm thinking of an example where we would need this method to update CORS setting and wonder how does it differ from setting it via CORS in the app config. Is it just for testing? An example would help ! 🙏

Reply:
In our use case, we want to use most of the options from the app config but we need a function option to resolve the origin dynamically. now we are using the `enableCors` method to pass all the configurations.

```ts
  const corsOptions: CorsOptions = {
    methods: corsMethods, // we need to pass this because the method overrides all the options
    credentials: corsCredentials, // same here
    origin: (origin, callback) => {
     // custom logic to allow dynamic origins
    },
  };

  const service = createServiceBuilder(module)
    .loadConfig(config)
    .addRouter('/', apiRouter)
    .enableCors(corsOptions);
```

after this change, we can leave the methods and credentials on the app config and only update the origin.

```ts
  const corsOptions: CorsOptions = { // rest of options can be updated from app config
    origin: (origin, callback) => {
     // custom logic to allow dynamic origins
    },
  };

  const service = createServiceBuilder(module)
    .loadConfig(config)
    .addRouter('/', apiRouter)
    . updateCorsOptions(corsOptions);
```

Reply:
Oh really helpful. Thanks Juan!